src/wp-admin/includes/media.php

```diff
@@ -547,13 +547,16 @@ function wp_iframe( $content_func, ...$args ) {
 		wp_enqueue_style( 'deprecated-media' );
 	}
 
+	ob_start();
 	?>
-	<script type="text/javascript">
+	<script>
 	addLoadEvent = function(func){if(typeof jQuery!=='undefined')jQuery(function(){func();});else if(typeof wpOnload!=='function'){wpOnload=func;}else{var oldonload=wpOnload;wpOnload=function(){oldonload();func();}}};
 	var ajaxurl = '<?php echo esc_js( admin_url( 'admin-ajax.php', 'relative' ) ); ?>', pagenow = 'media-upload-popup', adminpage = 'media-upload-popup',
 	isRtl = <?php echo (int) is_rtl(); ?>;
 	</script>
 	<?php
+	wp_print_inline_script_tag( trim( str_replace( array( '<script>', '</script>' ), '', ob_get_clean() ) ) );
```

Comment:
I'm not really happy with the boilerplate `ob_start()` followed by an inline script followed by:

```php
wp_print_inline_script_tag( trim( str_replace( array( '<script>', '</script>' ), '', ob_get_clean() ) ) );
```

I think ideally there would be a helper to do this automatically. Consider, for example, if `wp_print_inline_script_tag()` actually allowed a closure to be provided for the `$javascript` parameter in addition to a string. It could handle the output buffering automatically, for instance:

```php
wp_print_inline_script_tag( static function () {
	?>
	<script>
	addLoadEvent = function(func){if(typeof jQuery!=='undefined')jQuery(function(){func();});else if(typeof wpOnload!=='function'){wpOnload=func;}else{var oldonload=wpOnload;wpOnload=function(){oldonload();func();}}};
	var ajaxurl = '<?php echo esc_js( admin_url( 'admin-ajax.php', 'relative' ) ); ?>', pagenow = 'media-upload-popup', adminpage = 'media-upload-popup',
		isRtl = <?php echo (int) is_rtl(); ?>;
	</script>
	<?php
} );
```

When a closure is passed, it could automatically start and end output buffering, strip the `script` start and end tags, and trim whitespace. If the output buffer lacks a `<script>` it could issue a `_doing_it_wrong()`.

Reply:
The only purpose for having the `script` tags in the PHP code is to enable IDEs to do syntax-highlighting, syntax checking, autocompletion, etc. This is valuable so I think we should facilitate it somehow.

Reply:
This modification to `wp_get_inline_script_tag()` has since been reverted in this PR.